tests/unit/test_effect.py

```diff
@@ -18,37 +18,42 @@
 
 """This pytest module tries to test the correctness of the class Effect."""
 
-from palace import Context, Effect
+from palace import Effect, Source
 from pytest import raises
 
-from math import inf
 
-
-def test_gain(device):
+def test_gain(context):
     """Test write-only property `gain`."""
-    with Context(device), Effect() as fx:
+    with Effect() as fx:
         fx.gain = 0
         fx.gain = 1
-        fx.gain = 7/5
         fx.gain = 5/7
+        with raises(ValueError): fx.gain = 7/5
         with raises(ValueError): fx.gain = -1
 
 
-def test_send_auto():
+def test_send_auto(context):
     """Test write-only property `send_auto`."""
-    with Context(device), Effect() as fx:
+    with Effect() as fx:
         fx.send_auto = False
         fx.send_auto = True
-        with raises(ValueError): fx.gain = None
+        with raises(TypeError): fx.gain = None
+
+
+def test_source_sends(context):
+    """Test property `source_sends` by assigning it to a source."""
+    with Source() as src, Effect() as fx:
+        src.sends[0].effect = fx
```

Comment:
Looks like you forgot to test what's supposed to be tested lmao

Reply:
I just find out that `fx.source_sends[0]` and `src` are different object, with the same value 🤔 so if the user uses this to access the source, they will get the wrong instances.

Reply:
> different object, with the same value thinking so if the user uses this to access the source they will get the wrong instances.

Well yes but actually no: all functionalities we provide are from alure, and long as they refer to the same alure source, the result of calls (say, setting position) will be identical.  BTW could you please compare the tuples right-away, the `[0]` and `[1]` things make me uncomfortable.  Otherwise LGTM.

Reply:
Oh nvm, since they have the same impl, they would be the same anyway.